vita3k/renderer/src/vulkan/texture.cpp

```diff
@@ -92,19 +117,16 @@ void sync_texture(VKContext &context, MemState &mem, std::size_t index, SceGxmTe
 
         image = context.state.surface_cache.retrieve_color_surface_texture_handle(
             mem, width, height, stride_in_pixels, format_target_of_texture, surface_type, static_cast<bool>(texture.gamma_mode), Ptr<void>(data_addr),
-            renderer::SurfaceTextureRetrievePurpose::READING, swizzle);
+            renderer::SurfaceTextureRetrievePurpose::READING, swizzle, nullptr, nullptr, &texture_viewport);
     }
 
-    vk::ImageLayout layout = vk::ImageLayout::eShaderReadOnlyOptimal;
-    if (image) {
-        layout = vk::ImageLayout::eGeneral;
-    } else {
-        // Try to retrieve S24D8 texture
+    if (!image && is_depth_stencil_compatible_format(base_format)) {
+        // Try to retrieve depth/stencil
```

Comment:
Maybe add a warning if format is not compatible?

Reply:
No, this path is taken by all textures (except color surfaces). This is just an optimisation. For example, if the game tries to read a BC3 or RGBA16 texture, we know that it can't be a depth or stencil surface so no need to check if its address matches something in the depth-stencil cache.